Geometry/HGCalCommonData/plugins/dd4hep/DDHGCalModuleAlgo.cc

```diff
@@ -30,32 +30,32 @@ static long algorithm(dd4hep::Detector& /* description */, cms::DDParsingContext
   const auto& wafer = args.value<std::vector<std::string> >("WaferName");    // Wafers
   auto materials = args.value<std::vector<std::string> >("MaterialNames");   // Materials
   const auto& names = args.value<std::vector<std::string> >("VolumeNames");  // Names
-  const auto& thick = args.value<std::vector<double> >("Thickness");         // Thickness of the material
+  auto thick = args.value<std::vector<double> >("Thickness");                // Thickness of the material
   std::vector<int> copyNumber;                                               // Initial copy numbers
   for (unsigned int i = 0; i < materials.size(); ++i) {
     if (materials[i] == "materials:M_NEMAFR4plate")
       materials[i] = "materials:M_NEMA FR4 plate";
     copyNumber.emplace_back(1);
   }
 #ifdef EDM_ML_DEBUG
-  edm::LogVerbatim("HGCalGeom") << "DDHGCalModule: " << wafer.size() << " wafers";
+  edm::LogVerbatim("HGCalGeom") << "DDHGCalModuleAlgo: " << wafer.size() << " wafers";
   for (unsigned int i = 0; i < wafer.size(); ++i)
     edm::LogVerbatim("HGCalGeom") << "Wafer[" << i << "] " << wafer[i];
-  edm::LogVerbatim("HGCalGeom") << "DDHGCalModule: " << materials.size() << " types of volumes";
+  edm::LogVerbatim("HGCalGeom") << "DDHGCalModuleAlgo: " << materials.size() << " types of volumes";
   for (unsigned int i = 0; i < names.size(); ++i)
     edm::LogVerbatim("HGCalGeom") << "Volume [" << i << "] " << names[i] << " of thickness " << thick[i]
                                   << " filled with " << materials[i] << " first copy number " << copyNumber[i];
 #endif
-  const auto& layers = args.value<std::vector<int> >("Layers");             // Number of layers in a section
-  const auto& layerThick = args.value<std::vector<double> >("LayerThick");  // Thickness of each section
-  const auto& layerType = args.value<std::vector<int> >("LayerType");       // Type of the layer
-  const auto& layerSense = args.value<std::vector<int> >("LayerSense");     // Content of a layer (sensitive?)
+  const auto& layers = args.value<std::vector<int> >("Layers");          // Number of layers in a section
+  auto layerThick = args.value<std::vector<double> >("LayerThick");      // Thickness of each section
```

Comment:
why is this now a copy rather than const ref? It doesn't seem like the values are changed anywhere later.

Reply:
put back